src/proxy_app/telemetry/logger.py

```
@@ -0,0 +1,267 @@
+            if ttft == 0.0:
+                ttft = total
+        except Exception:
+            pass
+
+    # --- streaming first-token capture (async) ---
+    async def async_log_stream_event(self, kwargs, response_obj, start_time, end_time):
+        try:
+            rid = kwargs.get("litellm_call_id") or str(id(kwargs))
+            if rid in self._first_token_times and self._first_token_times[rid] == 0.0:
+                self._first_token_times[rid] = _now_ms()
+        except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)